plugins/com.google.cloud.tools.eclipse.dataflow.ui.test/src/com/google/cloud/tools/eclipse/dataflow/ui/launcher/PipelineArgumentsTabTest.java

```diff
@@ -61,35 +77,150 @@
 
     @Rule public ShellTestResource shellResource = new ShellTestResource();
 
+    private IWorkspaceRoot workspaceRoot;
+    private DataflowDependencyManager dependencyManager;
+    private PipelineArgumentsTab pipelineArgumentsTab;
+
+    private IProject project1;
+    private ILaunchConfigurationWorkingCopy configuration1;
+    private IProject project2;
+    private ILaunchConfigurationWorkingCopy configuration2;
+
+    @Before
+    public void setUp() throws CoreException {
+      workspaceRoot = mock(IWorkspaceRoot.class);
+      dependencyManager = mock(DataflowDependencyManager.class);
+
+      project1 = mockProject("project1");
+      when(workspaceRoot.getProject(eq("project1"))).thenReturn(project1);
```

Comment:
Nit: you don't need `eq` when all arguments are constants.

Reply:
`eq()` is necessary if you use pattern matchers, as we do in `testSuppressingUpdates()`

      when(workspaceRoot.getProject(anyString())).thenReturn(mock(IProject.class));